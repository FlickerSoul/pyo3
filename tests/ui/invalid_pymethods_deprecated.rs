use pyo3::prelude::*;

#[pyclass]
struct DeprecatedMethodContainer {}

#[pymethods]
impl DeprecatedMethodContainer {
    #[pyo3(deprecated = "deprecated __traverse__")]
    fn __traverse__(&self, _visit: pyo3::gc::PyVisit<'_>) -> Result<(), pyo3::PyTraverseError> {
        Ok(())
    }
}

#[pymethods]
impl DeprecatedMethodContainer {
    #[classattr]
    #[pyo3(deprecated = "deprecated class attr")]
    fn deprecated_class_attr() -> i32 {
        5
    }
}

fn main() {}
