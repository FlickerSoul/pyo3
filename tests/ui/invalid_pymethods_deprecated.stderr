error: __traverse__ cannot be used with #[pyo3(warn)] nor #[pyo3(deprecated)]
 --> tests/ui/invalid_pymethods_deprecated.rs:8:12
  |
8 |     #[pyo3(deprecated = "deprecated __traverse__")]
  |            ^^^^^^^^^^

error: #[classattr] cannot be used with #[pyo3(warn)] nor #[pyo3(deprecated)]
  --> tests/ui/invalid_pymethods_deprecated.rs:17:12
   |
17 |     #[pyo3(deprecated = "deprecated class attr")]
   |            ^^^^^^^^^^
